### ISSUE TITLE
Posterior predictive distribution

### ISSUE THREAD

**Issue by howardnewyork** [greta-dev/greta#177]
How do you generate a posterior predictive distribution after the mcmc function has been run?  E.g. I am trying to do something like this (see "This produces junk section" below).  Is it possible to simulate discrete random variables, once the mcmc function has been run, i.e. something similar to Stan's "rng" functions?

Thanks,
Howard

```
library(greta)

x <- c(1.6907, 1.7242, 1.7552, 1.7842, 1.8113, 1.8369, 1.8610, 1.8839)
n <- c(59, 60, 62, 56, 63, 59, 62, 60)
r <- c(6, 13, 18, 28, 52, 53, 61, 60)
N <- 8

alpha_star <- normal(0, 32)
beta <- normal(0, 32)
p <- ilogit(alpha_star + beta * (x - mean(x)))
distribution(r) <- binomial(n, p)

alpha <- alpha_star - beta * mean(x)
rhat <- p * n



mod = model(alpha_star, beta)
plot(mod)
draws= mcmc(mod, n_samples = 50, warmup = 30)
draws


#####################
#####################
# Simulate new draws assuming a normal approximation

#this works
calculate(rhat, draws)

# This produces junk
r_pred = normal(rhat, sd = greta_array(sqrt(n*p*(1-p)))) # Approximate with normal distribution
r_pred_calc = calculate(r_pred, draws)
r_pred_calc
#####################
#####################

```

**Comment by goldingn** [comment]
You can't (currently) add  new variables to a greta model after sampling and calculate the values. You can only use new operation and data greta arrays.

You'll have to extract the posterior samples for `rhat` and calculate `r_pred` manually. 

It would be really nice to be able to generate independent random samples of variables like this though. It could also be useful for automatic generation of initial values, since they could be sampled from their priors.

I won't have time to add that in the near future (just a bunch of other features higher up my to do list) but I'll leave this issue open until I do.

**Comment by howardnewyork** [comment]
Thanks for the speedy response!!! I am starting to learn Greta, and its really amazing.

**Comment by goldingn** [comment]
I've opened [a forum post](https://forum.greta-stats.org/t/feedback-requested-proposed-simulate-interface/154) asking for feedback on the proposed interface to a `simulate()` function that would enable this. @howardnewyork, please let me know your thoughts there if you have time!

**Comment by goldingn** [comment]
This is now implemented in master (PR #342), and there is an example that goes through what you can do. You just need to edit your penultimate line of code to:
```r
r_pred_calc = calculate(r_pred, values = draws, nsim = 200)
```
